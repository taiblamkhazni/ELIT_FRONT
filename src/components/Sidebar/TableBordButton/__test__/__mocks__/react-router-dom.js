/**
 * @file react-router-dom.js
 * @brief Fichier qui réexporte les modules de "react-router-dom" et simule la fonction "useNavigate" pour les tests.
 */
export * from "react-router-dom";

/**
 * Fonction jest qui simule "useNavigate" de "react-router-dom" pour les tests.
 *
 * @function useNavigate
 * @returns {function} Une fonction jest mockée.
 */
export const useNavigate = jest.fn();
