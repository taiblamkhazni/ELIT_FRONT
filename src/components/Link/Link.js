/**
 * @file Link.js
 * @brief This module exports StyledLink component
 */
import {Link as RouterLink} from "react-router-dom";
import styled from "styled-components"
import { base } from "theme/base"
/** Custom Style of a link Component */
export const StyledLink = styled(RouterLink)`
  & {
    text-decoration: none;
    color: ${base.colors.primaires.blue};
  }

  &:hover{
    text-decoration: none;
    color: ${base.colors.secondaires.blue};
}
`;
