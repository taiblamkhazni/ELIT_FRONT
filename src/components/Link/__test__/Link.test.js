/**
 * @file Link.test.js
 * @brief Ce fichier contient des tests pour les composant  BlockButton, ButtonNoBackground, LinkButton et NextStepButton.
 */
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import { StyledLink } from "../Link"

import "jest-styled-components"

test("StyledLink tests props active", () => {

  const tree = renderer.create(
    <MemoryRouter>
      <StyledLink />
    </MemoryRouter>).toJSON();

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule("text-decoration", "none")
  expect(tree).toHaveStyleRule("color", "#116E9C")
})

/*test('with props', () => {
    const { container } = render(
      <MemoryRouter>
        <StyledLink color="red" />
      </MemoryRouter>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule("text-decoration", "none")
    expect(container.firstChild).toHaveStyleRule("color", "red")
})*/
