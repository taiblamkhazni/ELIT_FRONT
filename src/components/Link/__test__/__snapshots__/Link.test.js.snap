// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink tests props active 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #116E9C;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6FC1E2;
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
/>
`;
